sweepbatcher/sweep_batch.go

```diff
@@ -215,6 +215,12 @@ type batch struct {
 	// reorgChan is the channel over which reorg notifications are received.
 	reorgChan chan struct{}
 
+	// testReqs is a channel where test requests are received.
+	// This is used only in unit tests! The reason to have this is to
+	// avoid data races in require.Eventually calls running in parallel
+	// to the event loop. See method testRunInEventLoop().
+	testReqs chan *testRequest
```

Comment:
I wonder if we could avoid this by instead using checkpoints in within mocks?

Reply:
Most of the checks that use this feature access `batcher.batches` and fields of a `batch`. The fields (e.g. fee rate, the list of sweeps) are updated in the main code, not in the mocks.